Cannot find module './device-config-schema'

error [pimatic]: Could not initialize the plugin "thinkingcleaner": Cannot find module './device-config-schema'

I just installed it to take a look at it in the GUI. I don't have a think cleaner device yet. So i don't know if this is a installation error or an cannot find device error?

--- a/package.json
+++ b/package.json
@@ -7,7 +7,8 @@
     "thinkingcleaner.coffee",
     "README.md",
     "thinkingcleaner-config-schema.coffee",
-    "device-config-schema.coffee"
+    "device-config-schema.coffee",
+    "app"
   ],
   "version": "0.8.2",
   "repository": {

--- a/package.json
+++ b/package.json
@@ -6,7 +6,8 @@
   "files": [
     "thinkingcleaner.coffee",
     "README.md",
-    "thinkingcleaner-config-schema.coffee"
+    "thinkingcleaner-config-schema.coffee",
+    "device-config-schema.coffee"
   ],
   "version": "0.8.2",
   "repository": {

--- a/test/roomba-mockup.js
+++ b/test/roomba-mockup.js
@@ -1,0 +1,23 @@
+var express = require('express');
+var app = express();
+var port = 3000;
+var status = "off";
+
+app.all("*", function (req, res) {
+    console.log("Request", req.url);
+
+    var match = req.url.match(/command=([A-Za-z]+)/);
+    if (match !== null) {
+        status = match[1];
+    }
+    res.send(JSON.stringify({
+        status: {
+            cleaner_state: status,
+            battery_charge: 60
+        }
+    }));
+});
+
+app.listen(port, function () {
+    console.log('Roomba mockup listening on port ' + port);
+});

--- a/thinkingcleaner.coffee
+++ b/thinkingcleaner.coffee
@@ -2,6 +2,9 @@ module.exports = (env) ->
 
   Promise = env.require 'bluebird'
   assert = env.require 'cassert'
+  _ = env.require 'lodash'
+  __ = env.require("i18n").__
+  M = env.matcher
 
   request = require 'request'
   Promise.promisifyAll(request)
@@ -14,6 +17,8 @@ module.exports = (env) ->
         configDef: deviceConfigDef.ThinkingCleanerDevice,
         createCallback: (config) -> new ThinkingCleanerDevice(config)
       })
+
+      @framework.ruleManager.addActionProvider(new ThinkingCleanerModeActionProvider(@framework))
 
       #wait till all plugins are loaded
       @framework.on "after init", =>
@@ -102,7 +107,7 @@ module.exports = (env) ->
       # Try to match the input string with:
       M(input, context)
         .match('set mode of ')
-        .matchDevice(thermostats, (next, d) =>
+        .matchDevice(tcleaners, (next, d) =>
           next.match(' to ')
             .matchStringWithVars( (next, ts) =>
               m = next.match(' mode', optional: yes)

--- a/thinkingcleaner.coffee
+++ b/thinkingcleaner.coffee
@@ -168,11 +168,17 @@ module.exports = (env) ->
       assert @valueTokens?
 
     _doExecuteAction: (simulate, value) =>
-      return (
+      new Promise( (resolve, reject) =>
         if simulate
-          Promise.resolve __("would set mode %s to %s%%", @device.name, value)
+          resolve __("would set mode %s to %s%%", @device.name, value)
         else
-          @device.sendCommand(value).then( => __("set mode %s to %s", @device.name, value) )
+          @device.sendCommand(value)
+          .then( =>
+            resolve __("set mode %s to %s", @device.name, value)
+          )
+          .catch( (error) =>
+            reject if error instanceof Error then error else new Error(error)
+          )
       )
 
     executeAction: (simulate) => 

--- a/thinkingcleaner.coffee
+++ b/thinkingcleaner.coffee
@@ -5,9 +5,8 @@ module.exports = (env) ->
   _ = env.require 'lodash'
   __ = env.require("i18n").__
   M = env.matcher
-
   request = require 'request'
-  Promise.promisifyAll(request)
+
 
   class ThinkingCleanerPlugin extends env.plugins.Plugin
 
@@ -57,34 +56,48 @@ module.exports = (env) ->
       @id = @config.id
       @name = @config.name
       @host = @config.host
+      @_state = "off"
+      @_battery = 0
       @interval = @config.interval
       super()
-      @readLoop    
+      @readLoop()
 
-    readLoop: =>
+    readLoop: ->
       setInterval( =>
-        request 'http://"+@host+"/status.json', (error, response, body) =>
+        request "http://"+@host+"/status.json", (error, response, body) =>
           if (!error && response.statusCode == 200)
             data = JSON.parse(body)
-            @setState data.status.cleaner_state 
-            @setBattery data.status.battery_charge
+            if data.status?
+              @setState data.status.cleaner_state
+              @setBattery data.status.battery_charge
       , @interval)
 
     getBattery: () -> Promise.resolve(@battery)
     getState: () -> Promise.resolve(@state)
 
     setBattery: (battery) ->
-      @battery = battery
-      @emit "battery", @battery
+      @_battery = battery
+      @emit "battery", @_battery
+
+    getBattery: () ->
+      Promise.resolve @_battery
 
     setState: (state) ->
-      @state = state
-      @emit "state", @state
+      @_state = state
+      @emit "state", @_state
+
+    getState: () ->
+      Promise.resolve @_state
 
     sendCommand: (command) ->
-      request "http://"+@host+"/command.json?command="+command, (error, response, body) =>
-        if (!error && response.statusCode == 200) 
-          data = JSON.parse(body)
+      new Promise( (resolve, reject) =>
+        request "http://"+@host+"/command.json?command="+command, (error, response, body) =>
+          if (!error && response.statusCode == 200)
+            data = JSON.parse(body)
+            resolve data
+          else
+            reject error
+      )
 
   class ThinkingCleanerModeActionProvider extends env.actions.ActionProvider
 
@@ -145,9 +158,9 @@ module.exports = (env) ->
     _doExecuteAction: (simulate, value) =>
       return (
         if simulate
-          __("would set mode %s to %s%%", @device.name, value)
+          Promise.resolve __("would set mode %s to %s%%", @device.name, value)
         else
-          @device.sendCommand(value).then( => __("set mode %s to %s%%", @device.name, value) )
+          @device.sendCommand(value).then( => __("set mode %s to %s", @device.name, value) )
       )
 
     executeAction: (simulate) => 

--- a/thinkingcleaner.coffee
+++ b/thinkingcleaner.coffee
@@ -56,6 +56,8 @@ module.exports = (env) ->
       @id = @config.id
       @name = @config.name
       @host = @config.host
+      # port member is used for testing purposes
+      @port = 80
       @_state = "off"
       @_battery = 0
       @interval = @config.interval
@@ -64,7 +66,7 @@ module.exports = (env) ->
 
     readLoop: ->
       setInterval( =>
-        request "http://"+@host+"/status.json", (error, response, body) =>
+        request "http://#{@host}:#{@port}/status.json", (error, response, body) =>
           if (!error && response.statusCode == 200)
             data = JSON.parse(body)
             if data.status?
@@ -90,13 +92,23 @@ module.exports = (env) ->
       Promise.resolve @_state
 
     sendCommand: (command) ->
+      switch command
+        when 'off' then (
+          command = 'poweroff'
+        )
+        when 'findme' then (
+          command = 'find_me'
+        )
       new Promise( (resolve, reject) =>
-        request "http://"+@host+"/command.json?command="+command, (error, response, body) =>
-          if (!error && response.statusCode == 200)
-            data = JSON.parse(body)
-            resolve data
+        request "http://#{@host}:#{@port}/command.json?command="+command, (error, response, body) =>
+          if not error?
+            if (response.statusCode == 200)
+              data = JSON.parse(body)
+              resolve data
+            else
+              reject "Unexpected HTTP response status code #{response.statusCode}"
           else
-            reject error
+            reject error.message ? error
       )
 
   class ThinkingCleanerModeActionProvider extends env.actions.ActionProvider

